### ISSUE TITLE
TorchVision support for pre-trained, quantised, ResNet50

### ISSUE THREAD

**Issue by nSircombe** [pytorch/vision#3362]
## 🚀 Feature
Addition of pre-trained, quantised ResNet50 model to `torchvision.models`

## Motivation

Quantised models have been shown to deliver greater performance than their fp32 equivalents without sacrificing accuracy. Expanding the portfolio of quantised models available to TorchVision users would provide functionality equivalent to what is currently available in other framworks.

## Pitch

Quantised, pre-trained, ResNet50 models are available for use on with TensorFlow (https://github.com/IntelAI/models/blob/master/benchmarks/image_recognition/tensorflow/resnet50v1_5/README.md#int8-inference-instructions) and typically deliver improved performance over their fp32 equivalents.

Are there any plans to introduce a similar INT8 ResNet50 model to TorchVision?

## Alternatives

Where the workload or benchmark requires ResNet50, the alternative is currently to use a different framework.

**Comment by oke-aditya** [comment]
Yes ! Torchvision does provide Quantized models, such as quantized `resnet50`. All the models are `int8` quantized.

Documentation for these is available over master. Visit [here](https://github.com/pytorch/vision/blob/master/docs/source/models.rst#quantized-models)

I think they are not reflected in [current website documentation,](https://pytorch.org/vision/stable/models.html#classification) but with next release it will be updated.

**Comment by nSircombe** [comment]
Hi @oke-aditya,

Thanks for your quick response. I'm sorry I should have been clearer in my original query, I'm curious about quantised ResNet50 models and accompanying support for a quantised backend. At present if I use `mobilenet_v2` I'm able to get a quantised model which uses the `QNNpack` backend (on AArch64)...

```
>>> import torchvision.models as models
>>> model = models.quantization.mobilenet_v2(pretrained=True, quantize=True)
```

However, this doesn't appear to be the case for ResNet50...

```
>>> import torchvision.models as models
>>> resnet50 = models.quantization.resnet50()
>>> resnet50 = models.quantization.resnet50(pretrained=True, quantize=True)
RuntimeError: Quantized backend not supported 
```

So it appears that support for a quantised backend is currently missing for ResNet50, is that the case? If so are there any plans to expand support?

**Comment by oke-aditya** [comment]
There are two Quantization backend in PyTorch `fbgemm` and `QNNpack`.
As you pointed out `mobilenet_v2` is available with `QNNpack`backend.

For `resne50t` model it is available with `fbgemm` backend. Have a look [here](https://github.com/pytorch/vision/blob/master/torchvision/models/quantization/resnet.py#L116)

I think you are very right that models are not available for both quantization backends. Also that some are available for `QNNpack` while some are for `fbgemm`.

**Comment by nSircombe** [comment]
Thanks @oke-aditya, I didn't find any details of QNNpack support for ResNet50 in the plans, but have asked in-thread.

I've updated my 'Feature Request' above in light of our discussion, so it's more specific to my use-case (QNNpack backend, on AArch64).

**Comment by fmassa** [comment]
Thanks for the request and sorry for the delay in replying.

I might not remember correctly anymore, but IIRC we provided `fbgemm` backends for ResNet because `fbgemm` was better suited for server inference, and ResNet50 was normally too big for running on embedded devices (on which qnnpack was better).

@nSircombe are you planning on running ResNet50 on embedded devices, or is it that fbgemm doesn't work in your system?
